lib/components/app/batch-routing-panel.tsx

```diff
@@ -12,14 +11,30 @@ import SwitchButton from '../form/switch-button'
 import UserSettings from '../form/user-settings'
 import ViewerContainer from '../viewers/viewer-container'
 
+interface Props {
+  activeSearch: any
+  intl: IntlShape
+  mobile?: boolean
+  showUserSettings: boolean
+}
+
 /**
  * Main panel for the batch/trip comparison form.
  */
-class BatchRoutingPanel extends Component {
-  handleSubmit = (e) => e.preventDefault()
+class BatchRoutingPanel extends Component<Props> {
+  state = {
+    planTripClicked: false
```

Comment:
Do we need to create a new state for this? Is it not possible to determine this info from the redux store? Also, does this handle cases where a trip is loaded in via url parameters?

Reply:
My understanding is that the user must have clicked the Plan Trip button first before the invalid states are activated, regardless of the values in the fields. It didn't make sense to me at this time to add a global redux state for that. I don't see a way to track that the Plan Trip button has been clicked from the existing redux state.

If a URL is entered in the browser with one of the locations missing, the initial `aria-invalid` will still be `false` for both locations, until the user clicks Plan Trip.

Reply:
That doesn't sound right to me but I trust you.